hicommonwealth/commonwealth
# [Experimental] Improved Thread page cards

## Possible issue
This will only work for the first proposal, even if multiple are linked. It will require a large refactoring because our proposal fetching is done inside a react hook, which cannot be for looped over (bad practice, could lead to unexpected consequences). If we require multiple regular proposal cards in the same thread, I can go ahead with the large refactor, but it would be better to break up into a separate PR.

## Link to Issue
Closes: #4708 

## Description of Changes
- Added functionality to make the cards collapsable
- Added Snapshot info and vote cards to thread view (broken up because could not fit in one screenshot:
<img width="286" alt="Screenshot 2023-08-15 at 2 15 30 PM" src="https://github.com/hicommonwealth/commonwealth/assets/14794654/3186e728-3d4e-46b0-9e0e-3f59a898a267">
<img width="299" alt="Screenshot 2023-08-15 at 2 15 42 PM" src="https://github.com/hicommonwealth/commonwealth/assets/14794654/d33307f3-15e1-49f4-8cfb-1664bc0a5172">

Added thread cards for all proposal types. Some examples:

Cosmos regular proposals:
<img width="276" alt="Screenshot 2023-08-22 at 11 48 33 AM" src="https://github.com/hicommonwealth/commonwealth/assets/14794654/cfd78cee-5903-4810-9736-408a368f84b9">

Aave regular proposals:
<img width="280" alt="Screenshot 2023-08-22 at 12 08 00 PM" src="https://github.com/hicommonwealth/commonwealth/assets/14794654/90ada5da-3a01-4302-8ac2-77dc1c37785d">


## Test Plan
- Go to dydx community, create a discussion, link proposal and snapshot proposal, make sure correct cards display in right side.
- Go to stargaze, create a discussion, link proposal, make sure correct cards display
